src/coprocessor/tracker.rs

```diff
@@ -107,6 +107,10 @@ impl Tracker {
         self.current_stage = TrackerState::ItemFinished;
     }
 
+    pub fn collect_storage_statistics(&mut self, storage_stats: Statistics) {
```

Comment:
Can we avoid introducing this new interface?

Reply:
I know we should trying to avoid introducing a new interface, but I don't know how to avoid it for now.
Our goal is to avoid the calls in each pair of `tracker.on_begin_item()` and `tracker.on_finish_item()` being in different threads, so we move them into the `poll()`. If we still pass storage statistics to `on_finish_item()`, we need to borrow the `handler` into the `tracker` and call `handler.collect_scan_statistics()` inside `poll()` too. But we can't borrow `handler` because it is borrowed by the `handler.handle_request()`.
Maybe we can move the storage-statistics-collecting-related code from `on_finish_item()` to the new-introduced `collect_storage_statistics()` method. And because this is the only place references `on_finish_item()`, I think it wouldn't affect too much other code.